kernel/syscall.c

```diff
@@ -3933,26 +3933,37 @@ static long _syscall(void* args_)
 
             BREAK(_return(n, ret));
         }
-        case SYS_myst_is_shared_crt:
+        case SYS_myst_get_fork_mode:
         {
-            bool* is_parent_of_fork = (bool*)x1;
-            bool* is_child_fork = (bool*)x2;
+            myst_fork_mode_t* fork_mode = (myst_fork_mode_t*)x1;
+            bool* is_parent_of_fork = (bool*)x2;
+            bool* is_child_fork = (bool*)x3;
             long ret = 0;
             myst_thread_t* process = myst_find_process_thread(thread);
 
             _strace(n, NULL);
-            if (!is_parent_of_fork || !is_child_fork)
+            if (!fork_mode || !is_parent_of_fork || !is_child_fork)
```

Comment:
Seeing this makes me think the ``struct myst_fork_info`` struct as a single parameter might be cleaner.

Reply:
next PR